### PR TITLE
Added Document Groups

### PR DESCRIPTION
**Pull request by star-nox**
Added a new parameter `documentGroups` so we can set document groups via webscrape too.

### CODE REVIEW

**Review by KastanDay — APPROVED** [review]
LGTM but I didn't test. If you merge, test thoroughly in prod (lol).

**Comment by KastanDay** [comment]
Actually, I think you're missing a few places where groups need to be added, like here: 
https://github.com/UIUC-Chatbot/crawlee/blob/main/src/api/crawlee.ts#L85
```ts
              // Asynchronously call the ingestWebscrape endpoint without awaiting the result
              if (html) {
                const ingestUrl = process.env.INGEST_URL;

                if (!ingestUrl) {
                  console.error('Error: INGEST_URL environment variable is not defined.');
                  return;
                }

                fetch(ingestUrl, {
                  "method": "POST",
                  "headers": {
                    "Accept": "*/*",
                    "Accept-Encoding": "gzip, deflate",
                    "Authorization": `Bearer ${process.env.BEAM_API_KEY}`,
                    "Content-Type": "application/json"
                  },
                  "body": JSON.stringify({
                    base_url: config.url,
                    url: request.loadedUrl,
                    readable_filename: title,
                    content: html,
                    course_name: config.courseName,
                    // s3_paths: s3Key,
                  })
                })
```

**Comment by KastanDay** [comment]
^^ Yeah I think that's the only missing place. Based on a code search for `fetch`.

CleanShot 2024-12-10 at 10 26 01@2x

File: src/api/crawlee.ts [hunk 1]
```diff
@@ -96,6 +96,7 @@ export async function crawl(rawConfig: Config) {
                     readable_filename: title,
                     content: html,
                     course_name: config.courseName,
+                    groups: config.documentGroups,
```
Comment (star-nox) [inline src/api/crawlee.ts:99]: @KastanDay the groups parameter is present here - ingest function for regular HTML content

File: src/api/uploadToS3.ts [hunk 2]
```diff
@@ -130,6 +130,7 @@ export async function ingestPdf(s3Key: string, courseName: string, base_url: str
         readable_filename: path.basename(s3Key),
         s3_paths: s3Key,
         course_name: courseName,
+        groups: documentGroups,
```
Comment (star-nox) [inline src/api/uploadToS3.ts:133]: Here is the one which handles PDF.

**Pull request merged**